libraries/plugins/witness/include/graphene/witness/witness.hpp

```diff
@@ -48,14 +48,7 @@ namespace block_production_condition
 class witness_plugin : public graphene::app::plugin {
 public:
    ~witness_plugin() {
-      try {
-         if( _block_production_task.valid() )
-            _block_production_task.cancel_and_wait(__FUNCTION__);
-      } catch(fc::canceled_exception&) {
-         //Expected exception. Move along.
-      } catch(fc::exception& e) {
-         edump((e.to_detail_string()));
-      }
+      plugin_shutdown();
```

Comment:
This means `plugin_shutdown()` will be called twice when shutting down. I don't think it's correct.

Reply:
Thanks ! Absolutely agree ! Sorry, will fix now by new method.